network/identity.go

```diff
@@ -87,13 +99,19 @@ func (i *identity) setup() {
 				return
 			}
 
-			if i.srv.numOpenSlots() == 0 {
+			if conn.Stat().Direction == network.DirOutbound && i.srv.numOpenSlots() == 0 {
+				i.srv.Disconnect(peerID, ErrNoAvailableSlots.Error())
+
+				return
+			}
+
+			if conn.Stat().Direction == network.DirInbound && i.srv.inboundConns() >= i.srv.maxInboundConns() {
 				i.srv.Disconnect(peerID, ErrNoAvailableSlots.Error())
 
 				return
 			}
```

Comment:
Please make this a separate method that will do the disconnect logic if the slots are missing for the specific connection direction

Reply:
If anybody reading this, this is the commit that fixes it:

https://github.com/0xPolygon/polygon-sdk/pull/326/commits/7fa13584e2cdc484c1189865c7b2a51bc7ecec5c